spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binding/ChannelBindingService.java

```diff
@@ -78,13 +73,10 @@ public ChannelBindingService(ChannelBindingServiceProperties channelBindingServi
 		ConsumerProperties consumerProperties =
 				this.channelBindingServiceProperties.getConsumerProperties(inputChannelName);
 		if (binder instanceof ExtendedPropertiesBinder) {
-			ExtendedPropertiesBinder extendedPropertiesBinder = (ExtendedPropertiesBinder) binder;
-			Object extension = extendedPropertiesBinder.getExtendedConsumerProperties(inputChannelName);
-			ExtendedConsumerProperties extendedConsumerProperties = new ExtendedConsumerProperties(extension);
-			BeanUtils.copyProperties(consumerProperties, extendedConsumerProperties);
-			consumerProperties = extendedConsumerProperties;
+			consumerProperties = ProducerConsumerPropertiesUtil.getExtendedConsumerProperties((ExtendedPropertiesBinder) binder,
```

Comment:
In the end, in both cases we are binding a channel based on its name, to a given destination. So at a first glance, I think that a call to `channelBindingService.bindProducer(channel, channelName);` should be enough, instead of copying in the code and calling the utility method.